app/components/autocomplete/at_mention/at_mention.js

```diff
@@ -96,8 +96,13 @@ export default class AtMention extends PureComponent {
         }
 
         // Server request is complete
-        if (requestStatus !== RequestStatus.STARTED &&
-            (inChannel !== this.props.inChannel || outChannel !== this.props.outChannel || teamMembers !== this.props.teamMembers)) {
+        if (
+            groups !== this.props.groups ||
```

Comment:
Yes, they would always be different objects. Same case with `inChannel` and `outChannel` as well.

`app/components/autocomplete/at_mention/index.js`
<img width="901" alt="Screenshot 2020-11-04 at 11 06 38 PM" src="https://user-images.githubusercontent.com/3680799/98147422-8196e180-1ef2-11eb-8574-bc46008073d8.png">

Reply:
@migbot Yes, It would be a different object everytime in case of `groups` but it's memoized one for `teamMembers`, `inChannel` & `outChannel`.

I have converted `groups` related function to Reselect one. Now objects are different only when there is a change in group data but not with every componentWillReceiveProps call.